internal/broker/broker.go

```diff
@@ -450,7 +452,7 @@ func (b *Broker) handleIsAuthenticated(ctx context.Context, session *sessionInfo
 		authInfo = authCachedInfo{Token: t, RawIDToken: rawIDToken}
 		authInfo.UserInfo, err = b.fetchUserInfo(ctx, session, &authInfo)
 		if err != nil {
-			return AuthDenied, errorMessage{Message: fmt.Sprintf("could not get user info: %v", err)}
+			return AuthDenied, errorMessage{Message: err.Error()}
```

Comment:
So I would like to change this to:

```golang
if err != nil {
    slog.Error(err)
    return AuthDenied, errorMessage{Message: "could not fetch user info"}
}
```

and not call `logAndRedactError` in `fetchUserInfo`.

Reply:
We tried this approach already. What ends up happening is that, if we redact the message too high in the chain, the error message isn't clear enough to be helpful. So that's why I decided to redact it only when it matters (which is in the private functions that handle sensitive data directly). Looking at the golden files should give you a good example of how the errors will look in the end.

Reply:
But you're only adding "could not fetch user info" to the message in `fetchUserInfo` as well. We know that's what's failing here as well, so I don't see why we should it add it further down the chain.

Reply:
This is mainly to avoid too much redundancy in the code (we would need to add a `slog.Error` call to every error handling branch). It's very similar to the pattern we use when decorating the error messages with `decorate`.

Reply:
I see that it removes a small bit of redundancy, but I think that not adding a layer of indirection here actually makes the code more readable and makes it more obvious which messages we're returning to the user.

Reply:
I think it's also cleaner if we don't even have to use an `err` which we got from another function for the message which we show the user.